### PR TITLE
Images upload fixes

### PR DESCRIPTION
**Pull request by asaunier**
* Only moderators can upload copyright-typed images => see https://github.com/c2corg/v6_ui/issues/1272
* Client-side resizing was active only for the initial upload panel, not for the additional panel shown after a first upload has been triggered => there was no resizing in that case => perhaps causing https://github.com/c2corg/v6_ui/issues/929#issuecomment-268836270
* Save/Close buttons are disabled after clicking the Save button => prevent multiple requests, see https://github.com/c2corg/v6_ui/issues/1112 It would be nice to show a loading indicator too but not sure we can really use the ``appLoading`` directive here since we don't want to trigger the loading indicator when uploading files, only when saving the images data to the API.

### CODE REVIEW

**Comment by asaunier** [comment]
faf24da fixes the way to figure out if a document is collaborative => used for instance to figure out the default image types - collaborative or personal - of uploaded images.

File: c2corg_ui/static/js/imageuploader.js [hunk 1]
```diff
@@ -409,6 +417,23 @@ app.ImageUploaderController.prototype.resizeIf = function(
 };
 
 
+/**
+ * @param {Array.<string>} image_types
+ * @return {Array.<string>}
+ * @export
+ */
+app.ImageUploaderController.prototype.filterImageTypes = function(image_types) {
```
Comment (tsauerwein) [inline c2corg_ui/static/js/imageuploader.js]: camel case: `imageTypes`?

File: c2corg_ui/static/partials/imageuploader.html [hunk 2]
```diff
@@ -113,6 +113,8 @@
          ngf-keep="'distinct'"
          ngf-fix-orientation="true"
          ngf-allow-dir="true"
+         ngf-resize-if="uplCtrl.resizeIf($file, $width, $height)"
+         ngf-resize="{width: 2048, height: 2048, quality: 0.9}"
```
Comment (tsauerwein) [inline c2corg_ui/static/partials/imageuploader.html]: Do you maybe want to move these options into a variable in the controller and use `ngf-resize="uplCtrl.resizeOptions"` here and above?

File: c2corg_ui/static/js/imageuploader.js [hunk 3]
```diff
@@ -477,7 +477,9 @@ app.ImageUploaderModalController.prototype.close = function() {
  * @export
  */
 app.ImageUploaderModalController.prototype.save = function() {
+  $('.img-upl-button').addClass('disabled');
```
Comment (tsauerwein) [inline c2corg_ui/static/js/imageuploader.js]: Adding and removing the class like this with jQuery is not very Angular!

Comment (asaunier) [inline c2corg_ui/static/js/imageuploader.js]: I think I have copied some code found somewhere else in the UI :P 
How would you recommend to do it?

Comment (tsauerwein) [inline c2corg_ui/static/js/imageuploader.js]: More Angular-like would be to use [ng-class](https://docs.angularjs.org/api/ng/directive/ngClass) and to store the disabled status in a variable on the controller.

Comment (asaunier) [inline c2corg_ui/static/js/imageuploader.js]: Thanks. I will make changes to take your suggestions into account.

**Comment by asaunier** [comment]
@tsauerwein I have made the changes you have suggested. By the way I have realized that setting class ``disabled`` on the buttons does not prevent them to be clicked => I have also used the saving status to figure out whether their handlers should be enabled or not.

I will merge the PR when Travis is OK if you have no other comment.

**Pull request merged**